### ISSUE TITLE
[Question] Deleted messages older than 48hours

### ISSUE THREAD

**Issue by typechip** [telegramdesktop/tdesktop#23894]
Hi everyone, 

I recently deleted by mistake some messages when trying to backup them. I run a script that clicked made manual "Delete for everyone" actions in those messages.

The messages were older than 48 hours and I have read that those messages are not deleted, they are still in the cloud servers and so they are still visible for the rest of members of the group or conversation where they were sent.

So my question is: Do you know if there is a way to display those messages again? Do you think it is something doable in `tdesktop` worth exploring?

Thank you in advance!

**Comment by RememberTheAir** [comment]
Messages sent in private chats and in normal groups ([normal = non-supergroup](https://telegra.ph/groups-08-26)) are stored once for every chat party (that is, twice for private chats, and up to 200 times in normal group chats). Every accounts owns a copy of the message in its own cloud storage. If you delete your copy of the message without asking Telegram to delete everyone else's copy, the message is gone from your cloud storage but is still in the other chat participants' cloud storage. But you can't access their cloud storage, unless you login as them. There is no way to retrieve these messages and "copy" them back to your chat history

**Comment by typechip** [comment]
Thanks for your answer @RememberTheAir 

Do you know if the deleted messages could be stored somewhere in the cache of the `tdesktop` app? I have it installed in an ubuntu OS. I did a first search but didn't manage to get anything.

**Comment by john-preston** [comment]
@typechip No, tdesktop doesn't have messages cache, it loads them each launch from the server.

**Comment by typechip** [comment]
Thanks a lot for your answers.

@john-preston or @RememberTheAir , wouldn't you happen to know if there is an option in Telegram to recover the deleted personal history? maybe by contacting support?

Thank you.

**Comment by john-preston** [comment]
@typechip There is no such option, sorry.
